### PR TITLE
Start using fetch for the HTTP unit tests.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14114)

### CODE REVIEW

File: tests/unit/net/http_loader.rs [hunk 1]
```diff
@@ -401,32 +394,41 @@ fn test_check_default_headers_loaded_in_every_request() {
     let mut en_us: LanguageTag = Default::default();
     en_us.language = Some("en".to_owned());
     en_us.region = Some("US".to_owned());
-    let mut en: LanguageTag = Default::default();
-    en.language = Some("en".to_owned());
```
Comment (jdm) [inline tests/unit/net/http_loader.rs]: We should modify https://dxr.mozilla.org/servo/source/components/net/fetch/methods.rs#124 to call https://dxr.mozilla.org/servo/source/components/net/http_loader.rs#416 instead.

**Review by jdm — DISMISSED** [review]

**Comment by Ms2ger** [comment]
Fixed a test that expected the fetch version of `Accept-Language`.

**Pull request merged**